spatialmath/pose.go

```diff
@@ -97,11 +97,18 @@ func Compose(a, b Pose) Pose {
 }
 
 // PoseBetween returns the difference between two dualQuaternions, that is, the dq which if multiplied by one will give the other.
+// Example: if PoseBetween(a, b) = c, then Compose(a, c) = b.
 func PoseBetween(a, b Pose) Pose {
+	return Compose(PoseInverse(a), b)
+}
+
+// PoseBetweenInverse returns an origin pose which when composed with the first parameter, yields the second.
+// Example: if PoseBetweenInverse(a, b) = c, then Compose(c, a) = b.
+func PoseBetweenInverse(a, b Pose) Pose {
```

Comment:
This is unused and I think not worth keeping around if we don't need it as it adds complexity.  Also, unless I'm misunderstanding the same functionality is acheived by PoseBetween(b, a)

Reply:
It's a convenience function that could be useful; I would argue that as it's currently unused it doesn't add any complexity, and re-deriving the dual quaternion math it uses if we need to bring it back is non-trivial.

Due to the order-dependent nature of spatial operations, this is functionality that is in no way provided by any invocation of `PoseBetween`. PoseBetween(a, b) is syntactic sugar for `Compose(PoseInverse(a), b)`, while PoseBetweenInverse(a, b) is syntactic sugar for `Compose(b, PoseInverse(a))`